[enhancement] track last seen stepchart in EditMode, too

quietly-turning opened the pull request:
# About

This builds on #251 by allowing ScreenEditMenu to initialize into the last seen stepchart as well as the last seen song.

## Implementation

There are three new ThemePref values: 
 * `EditModeLastSeenStepsType` - a string like "StepsType_Dance_Single"
 * `EditModeLastSeenStyleType` - a string like "StyleType_OnePlayerOneSide"
 * `EditModeLastSeenDifficulty` - a string like "Difficulty_Challenge"

These are needed to configure GAMESTATE appropriately  and set the stepchart.

The code for getting and setting and validating these ThemePrefs grew quite a bit, so I moved it to a separate file and `LoadActor`'d that into `ScreenEditMenu underlay`.

## Considerations

I validate and set the song independently from the stepchart.

It's possible to switch games (i.e. from `dance` to `techno`) which would invalidate the `EditModeLastSeenStepsType` but not necessarily `EditModeLastSeenSong`, and I figured it's better to at least set the song rather than both-or-none.

With regard to "bad" data being retrieved from ThemePrefs, I've tested:
  * empty strings
  * garbage values
  * switching between stepmania games (dance to techno to pump)
  * mismatched StyleType and StepsType

## Miscellaneous

During testing, I found that the _fallback theme's `ThemePrefs.Get()` seems to choke if the string being read includes square brackets (i.e. `/Tachyon Epsilon/The Skies Are Not Enough - [Mad Matt]` and returns an empty string.  This is a problem, but outside the scope of this pull request.

quietly-turning committed:
track last seen stepchart in EditMode, too

2601a81 gave ScreenEditMenu the ability to initialize with the "last
seen" song from the previous Edit Mode session (rather than the first
song in the first pack, alphabetically).

This builds on that by allowing ScreenEditMenu to initialize into the
last seen stepchart as well.  It introduces three new ThemePref values.

I've tested:
  empty strings
  garbage values
  switching between stepmania games (dance to techno)
  mismatched StyleType and StepsType

teejusb commented on BGAnimations/ScreenEditMenu underlay/LastSeenSong.lua:
```diff
@@ -0,0 +1,111 @@
+-- ----------------------------------------------------------------
+
+local song_str      = ThemePrefs.Get("EditModeLastSeenSong")
+local diff_str      = ThemePrefs.Get("EditModeLastSeenDifficulty")
```
How does this handle files with multiple edits? Does it simply pick the first one?

quietly-turning replied:
> Does it simply pick the first one?

Yes.

quietly-turning committed:
show Description for Edit stepcharts in EditMenu

Simfiles can have multiple multiple "Edit" stepcharts, and
ScreenEditMenu previously didn't indicate which was which very clearly.
This changes the behavior so that an Edit stepchart's Description will
appear in Row4.

Using Tachyon Alpha/DeltaMAX as an example, where you might have
previously seen:
Hard                   9
Expert                15
Edit                  15
Edit                  14

You'd now now see:
Hard                   9
Expert                15
Edit   Corner Drill   15
Edit   Staircase      14

quietly-turning commented:
I've added Descriptions for Edit stepcharts to Row4 on ScreenEditMenu.

Using Tachyon Alpha/DeltaMAX as an example, where you might have previously seen:
<table>
<tr><td>Hard</td><td>9</td></tr>
<tr><td>Expert</td><td>15</td></tr>
<tr><td>Edit</td><td>14</td></tr>
</table>

<table>
<tr><td>Hard</td><td>9</td></tr>
<tr><td>Expert</td><td>Corner Dill 15</td></tr>
<tr><td>Edit</td><td>Staircase 14</td></tr>
</table>

https://user-images.githubusercontent.com/1253483/112526543-71d04a80-8d78-11eb-9b5a-da43927bead5.mov

I don't have a practical fix in mind for storing and setting the correct Edit stepchart but am open to suggestions.  This^ change seems good, regardless.

teejusb commented:
I think the PR looks pretty good to me.
> I don't have a practical fix in mind for storing and setting the correct Edit stepchart but am open to suggestions

I'm not sureee if it's possible since afaict `GetOneSteps` doesn't really let you set a specific Edit chart. I think the current  functionality is pretty acceptable.

natano reviewed (approved):
LGTM. Thanks for your work, @quietly-turning!
